### ISSUE TITLE
Display the tail call position information in OCaml

### ISSUE THREAD

**Issue by Khady** [freebroccolo/vscode-reasonml#62]
It seems that the `tail call` information given by merlin is not displayed (or I missed it)

Simple test case:

```ocaml
let rec test () =
  if 1 + 2 = 3 then
    test ()
  else
    (<cursor>)
```

The displayed type is `unit` and I don't see `tail call` anywhere.

I searched a little bit and found those related commits and issues:

- https://github.com/freebroccolo/vscode-reasonml/issues/8
- https://github.com/freebroccolo/vscode-reasonml/commit/f17b20e54d38834c283192e404fd2146e13e99c6
- https://github.com/freebroccolo/vscode-reasonml/blob/b86160c01cce3992cdc3142d38bb74b450374a51/src/shared/merlin/data.ts#L184-L207

I tried to play with `syntaxes/ocaml-hover-{type,signature}.json` to make them look like the `syntaxes/reason-hover-{type,signature}.json` files but without success.

**Comment by freebroccolo** [comment]
Yeah, originally I had it display that information. As you can see from those links, I disabled it at some point. I wasn't really happy with how it worked at the time. I can probably go back and try to make it work more along the lines of how I described it in #8.

I think there is some overlap in how this needs to be handled and #4. There, I had already implemented a little parser that would traverse the syntax from the current location up to the nearest head position of an application. I think I could reuse that with a few modifications to provide more accurate `tail call` information on hover.

**Comment by Khady** [comment]
Some screenshots of how emacs works with merlin. I don't know if it's what you are aiming for, but I find the behavior not too surprising.

![screenshot from 2017-05-30 04-59-41](https://cloud.githubusercontent.com/assets/974142/26568702/13ffc0b8-44f5-11e7-9577-cbc5751aa9c2.png)
![screenshot from 2017-05-30 04-59-34](https://cloud.githubusercontent.com/assets/974142/26568704/140320a0-44f5-11e7-9a12-a7e3694bf0a0.png)
![screenshot from 2017-05-30 04-59-24](https://cloud.githubusercontent.com/assets/974142/26568703/14012b2e-44f5-11e7-9e9c-682d1806d8c5.png)
![screenshot from 2017-05-30 04-59-18](https://cloud.githubusercontent.com/assets/974142/26568701/13f22368-44f5-11e7-9715-e732b46363e4.png)
